src/v_video.c

```diff
@@ -919,36 +919,39 @@ void V_PutBlock(int x, int y, int width, int height, byte *src)
 
 void V_ShadeScreen(void)
 {
-  int x, y;
-  byte *dest = dest_screen;
-  const int targshade = 20, step = 2;
-  static int oldtic = -1;
-  static int screenshade;
-
-  // [FG] start a new sequence
-  if (gametic - oldtic > targshade / step)
-  {
-    screenshade = 0;
-  }
+    int x, y;
+    static int oldtic = -1;
+    static int screenshade;
 
-  for (y = 0; y < video.height; y++)
-  {
-    for (x = 0; x < video.width; x++)
+    #define SHADE_TARGET 12
+    #define SHADE_STEP   4
+
+    // [FG] start a new sequence
+    if (gametic - oldtic > SHADE_TARGET / SHADE_STEP)
```

Comment:
Well, if we rush through the animation in 1/10 second, we could just as well drop it entirely.

Reply:
I agree 😄 I thought about compromise.